From 87601262394ed10813acd972d72bac392a0b76c5 Mon Sep 17 00:00:00 2001
From: Amir Vadai <avadai@redhat.com>
Date: Thu, 11 Sep 2014 12:00:35 -0400
Subject: [infiniband] mlx4: Make buffer larger to avoid overflow warning

Message-id: <71f474d3a68d98917cb8378e7376a0da7c5a7fab.1410425016.git.avadai@redhat.com>
Patchwork-id: 93761
O-Subject: [Patch RHEL7.1 V1 099/197] mlx4_core: Make buffer larger to avoid overflow warning
Bugzilla: 1107617 1107618 1107619 1107620
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>

From: Dan Carpenter <dan.carpenter@oracle.com>

My static checker complains that the sprintf() here can overflow.

 drivers/infiniband/hw/mlx4/main.c:1836 mlx4_ib_alloc_eqs()
 error: format string overflow. buf_size: 32 length: 69

This seems like a valid complaint.  The "dev->pdev->bus->name" string
can be 48 characters long.  I just made the buffer 80 characters instead
of 69 and I changed the sprintf() to snprintf().

Signed-off-by: Dan Carpenter <dan.carpenter@oracle.com>
Signed-off-by: Roland Dreier <roland@purestorage.com>
(cherry picked from commit 4661bd798f1b58ee2755bfa04003638345802680)
Signed-off-by: Amir Vadai <avadai@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/infiniband/hw/mlx4/main.c b/drivers/infiniband/hw/mlx4/main.c
index 59fd96c..c21c4ce 100644
--- a/drivers/infiniband/hw/mlx4/main.c
+++ b/drivers/infiniband/hw/mlx4/main.c
@@ -1803,7 +1803,7 @@ static void init_pkeys(struct mlx4_ib_dev *ibdev)
 
 static void mlx4_ib_alloc_eqs(struct mlx4_dev *dev, struct mlx4_ib_dev *ibdev)
 {
- char name[32];
+ char name[80];
  int eq_per_port = 0;
  int added_eqs = 0;
  int total_eqs = 0;
@@ -1833,8 +1833,8 @@ static void mlx4_ib_alloc_eqs(struct mlx4_dev *dev, struct mlx4_ib_dev *ibdev)
  eq = 0;
  mlx4_foreach_port(i, dev, MLX4_PORT_TYPE_IB) {
   for (j = 0; j < eq_per_port; j++) {
-   sprintf(name, "mlx4-ib-%d-%d@%s",
-    i, j, dev->pdev->bus->name);
+   snprintf(name, sizeof(name), "mlx4-ib-%d-%d@%s",
+     i, j, dev->pdev->bus->name);
    /* Set IRQ for specific name (per ring) */
    if (mlx4_assign_eq(dev, name, NULL,
         &ibdev->eq_table[eq])) {
-- 
1.7.1